src/main/java/seedu/address/model/person/ContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.model.person;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COMMENT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COUNTRY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STATUS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person}'s {@code Attribute} matches the corresponding keyword given.
+ */
+public class ContainsKeywordsPredicate<T> implements Predicate<Person> {
+
+    private final Prefix prefix;
+    private final Optional<T> keywords;
+
+    /**
+     * @param prefix of the corresponding field.
+     * @param keywords that the test is run against.
+     */
+    public ContainsKeywordsPredicate(Prefix prefix, Optional<T> keywords) {
+        this.prefix = prefix;
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        if (prefix.equals(PREFIX_NAME)) {
+            Optional<Name> name = (Optional<Name>) keywords;
+            return name.isEmpty() || person.getName().fullName.contains(name.get().fullName);
+        } else if (prefix.equals(PREFIX_PHONE)) {
+            Optional<Phone> phone = (Optional<Phone>) keywords;
+            return person.getPhone().value.equals(phone.orElse(person.getPhone()).value);
+        } else if (prefix.equals(PREFIX_EMAIL)) {
+            Optional<Email> email = (Optional<Email>) keywords;
+            return person.getEmail().value.equals(email.orElse(person.getEmail()).value);
+        } else if (prefix.equals(PREFIX_COUNTRY)) {
+            Optional<Country> country = (Optional<Country>) keywords;
+            return person.getCountry().value.equals(country.orElse(person.getCountry()).value);
+        } else if (prefix.equals(PREFIX_STATUS)) {
+            Optional<Status> status = (Optional<Status>) keywords;
+            return person.getStatus().value.equals(status.orElse(person.getStatus()).value);
+        } else if (prefix.equals(PREFIX_COMMENT)) {
+            Optional<Comment> comment = (Optional<Comment>) keywords;
+            return person.getComment().value.contains(comment.orElse(new Comment("")).value);
```

Comment:
no getCommand method for person class is it?

Reply:
On further thought, this also works; in fact, it also works for Name which removes the redundant isEmpty() check. Thank for pointing this out!